src/server/worker/datum/datum.go

```diff
@@ -416,3 +420,36 @@ func NewDeleter(metaFileWalker fileWalkerFunc, metaOutputClient, pfsOutputClient
 		return nil
 	}
 }
+
+// TODO: This should be removed when CopyFile is a part of ModifyFile.
+type datumClient struct {
+	*client.ModifyFileClient
+	pachClient *client.APIClient
+	commit     *pfs.Commit
+}
+
+func NewClient(mfc *client.ModifyFileClient, pachClient *client.APIClient, commit *pfs.Commit) Client {
+	return &datumClient{
+		ModifyFileClient: mfc,
+		pachClient:       pachClient,
+		commit:           commit,
+	}
+}
+
+func (dc *datumClient) CopyFile(dst string, srcFile *pfs.File, tag string) error {
+	return dc.pachClient.CopyFile(srcFile.Commit.Repo.Name, srcFile.Commit.ID, srcFile.Path, dc.commit.Repo.Name, dc.commit.ID, dst, false, tag)
+}
+
+type datumClientFileset struct {
+	*client.CreateFilesetClient
+}
+
+func NewClientFileset(cfc *client.CreateFilesetClient) Client {
+	return &datumClientFileset{
+		CreateFilesetClient: cfc,
+	}
+}
+
+func (dcf *datumClientFileset) CopyFile(_ string, _ *pfs.File, _ string) error {
+	panic("attempted copy file in fileset datum client")
```

Comment:
Should this be a panic rather than just returning an error? It seems like a panic might needlessly crash the whole program and any code calling this should be designed to handle an error when copying a file.

Reply:
This would be a critical issue, so a panic might be warranted since it gives a clear indication where the problem is and it is not recoverable. @brendoncarroll and I have debated panics vs returning errors for critical issues in the past and I warmed up to the idea of using panics, but if you feel differently then maybe we should discuss this decision. For this particular case, this is going to be very short lived because the `datum.Client` is completely removed in a PR I am about to put up, so this code is going away.

Reply:
There is no way the worker could be tricked into calling this? It's always a bug when it's called?

Reply:
Yeah, it should not be called, thus the panic. This code is gone in: https://github.com/pachyderm/pachyderm/pull/5879.

Reply:
If the code is gone then probably not worth worrying about this particular issue. In general I think `panic`s only make sense when it's an issue that will completely prevent the whole program from doing its job and restarting the pod is the only thing that might fix it.